src/storage/http/StorageHttpDownloadHandler.cpp

```diff
@@ -41,35 +43,88 @@ void StorageHttpDownloadHandler::init(nebula::hdfs::HdfsHelper *helper,
+        auto dirs = fs::FileUtils::listAllDirsInDir(root.c_str(), true);
+        for (auto& dir : dirs) {
+            if (fs::FileUtils::fileLastUpdateTime(dir.c_str()) <
+                time::WallClock::fastNowInSec() - 60 * 60 * 24 * FLAGS_download_keep_day) {
+                LOG(INFO) << "rm download dir: " << dir;
+                fs::FileUtils::remove(dir.c_str(), true);
+            }
+        }
+    }
+}
+
 void StorageHttpDownloadHandler::onRequest(std::unique_ptr<HTTPMessage> headers) noexcept {
     if (headers->getMethod().value() != HTTPMethod::GET) {
         // Unsupported method
         err_ = HttpCode::E_UNSUPPORTED_METHOD;
         return;
     }
-
-     if (!headers->hasQueryParam("host") ||
-         !headers->hasQueryParam("port") ||
-         !headers->hasQueryParam("path") ||
-         !headers->hasQueryParam("parts") ||
-         !headers->hasQueryParam("space")) {
-         LOG(ERROR) << "Illegal Argument";
-         err_ = HttpCode::E_ILLEGAL_ARGUMENT;
-         return;
-     }
-
-     hdfsHost_ = headers->getQueryParam("host");
-     hdfsPort_ = headers->getIntQueryParam("port");
-     hdfsPath_ = headers->getQueryParam("path");
-     partitions_ = headers->getQueryParam("parts");
-     spaceID_ = headers->getIntQueryParam("space");
-
-     for (auto &path : paths_) {
-         auto downloadPath = folly::stringPrintf("%s/nebula/%d/download", path.c_str(), spaceID_);
-         if (fs::FileUtils::fileType(downloadPath.c_str()) == fs::FileType::NOTEXIST) {
-             fs::FileUtils::makeDir(downloadPath);
-         }
-     }
+    if (!headers->hasQueryParam("host") ||
+        !headers->hasQueryParam("port") ||
+        !headers->hasQueryParam("path") ||
+        !headers->hasQueryParam("parts") ||
+        !headers->hasQueryParam("space")) {
+        LOG(ERROR) << "Illegal Argument";
+        err_ = HttpCode::E_ILLEGAL_ARGUMENT;
+        return;
+    }
+    hdfsHost_ = headers->getQueryParam("host");
+    hdfsPort_ = headers->getIntQueryParam("port");
+    hdfsPath_ = headers->getQueryParam("path");
+    auto existStatus = helper_->exist(hdfsHost_, hdfsPort_, hdfsPath_);
+    if (!existStatus.ok()) {
+        LOG(ERROR) << "Run Hdfs Test failed. " << existStatus.status().toString();
+        err_ = HttpCode::E_ILLEGAL_ARGUMENT;
+    }
+    bool exist = existStatus.value();
+    if (!exist) {
+        LOG(ERROR) << "Hdfs non exist. hdfs://" << hdfsHost_ << ":" << hdfsPort_ << hdfsPath_;
+        err_ = HttpCode::E_ILLEGAL_ARGUMENT;
+        return;
+    }
+    spaceID_ = headers->getIntQueryParam("space");
+    auto partitions = headers->getQueryParam("parts");
+    folly::split(",", partitions, parts_, true);
+    if (parts_.empty()) {
+        LOG(ERROR) << "Partitions should be not empty";
+        err_ = HttpCode::E_ILLEGAL_ARGUMENT;
+        return;
+    }
+    if (headers->hasQueryParam("tag")) {
+        auto& tag = headers->getQueryParam("tag");
+        tag_.assign(folly::to<TagID>(tag));
```

Comment:
folly::to maybe throw exceptions, we should catch them, or use tagName directly.

Reply:
yes

Reply:
Please fix the comments